lib/main_common.pm

```diff
@@ -2830,8 +2834,8 @@ sub load_ha_cluster_tests {
     # Show HA cluster status *before* fencing test and execute fencing test
     loadtest 'ha/fencing';
 
-    # Node1 will be fenced, so we have to wait for it to boot
-    boot_hdd_image if !get_var('HA_CLUSTER_JOIN');
+    # Node1 will be fenced, so we have to wait for it to boot. On svirt load only boot_to_desktop
+    check_var('BACKEND', 'svirt') ? loadtest 'boot/boot_to_desktop' : boot_hdd_image if !get_var('HA_CLUSTER_JOIN');
```

Comment:
Maybe `boot_to_desktop` can be used for all tests? But this is something that can be tested/changed in another PR.

I remember that I had issue with `boot_to_desktop` at this stage, but things have changed a lot since then.

Reply:
Will look into it.

Reply:
As it is, `boot_hdd_image` does handle the s390x over zKVM boot with `BACKEND=svirt` as can be seen in L2713, the problem with using that method here in L2838 is that it also schedules `installation/bootloader_s390` which is required at the start of the test, but not during a reboot.

`installation/bootloader_s390` prepares everything to start the zKVM VM, so during a reboot it will attempt to issue a `qemu-system-s390x` with write access to the same backing image that's already running; this causes the test to abort as the second `qemu-system-s390x` cannot get write access to the file as there is already a `qemu-system-s390x` process with write access to the backing image.

However, a cleaner approach could be to drop the call to `boot_hdd_image` in L2838 in favor of only a `loadteast 'boot/boot_to_desktop'` for all ARCHs and backends. Current [x86_64](https://openqa.suse.de/tests/3396310) and [ppc64le](https://openqa.suse.de/tests/3395194) tests over qemu are calling `boot_hdd_image`, but this method only schedules `boot/boot_to_desktop` in those scenarios.

The problem with going with only `boot/boot_to_desktop` could be when `ARCH=aarch64`, as `boot_hdd_image` also schedules `boot/uefi_bootmenu` in that [achitecture](https://openqa.suse.de/tests/3305984), so in short:

During boot we need:

1. `boot/boot_to_desktop` in qemu/x86_64
2. `boot/boot_to_desktop` in qemu/ppc64le
3. `boot/uefi_bootmenu` and `boot/boot_to_desktop` in qemu/aarch64
4. `installation/bootloader_zkvm` and `boot/boot_to_desktop` in svirt/x390x

During reboot we need:

1. `boot/boot_to_desktop` in qemu/x86_64
2. `boot/boot_to_desktop` in qemu/ppc64le
3. `boot/uefi_bootmenu` and `boot/boot_to_desktop` in qemu/aarch64
4. `boot/boot_to_desktop` in svirt/x390x

To be able to make `boot_hdd_image` to schedule `installation/bootloader_zkvm` on the initial boot and not after the fence, could be tricky TBH.